File: src/main/java/com/hrms/hr_payroll_management_system/service/BillingService.java
package com.hrms.hr_payroll_management_system.service;

import com.hrms.hr_payroll_management_system.dto.request.billing.ChangePlanRequest;
import com.hrms.hr_payroll_management_system.dto.response.billing.*;

import java.util.List;

public interface BillingService {
    SubscriptionResponse getSubscription();
    List<PlanResponse> getPlans();
    SubscriptionResponse changePlan(ChangePlanRequest request);
    SubscriptionResponse cancelSubscription();
    List<InvoiceResponse> getInvoices();
    BillingSummaryResponse getSummary();
}

File: frontend/src/services/Billing.service.js
import { billingApi } from '../api/billing.api';

export const billingService = {
  async subscription() {
    const res = await billingApi.getSubscription();
    return res.data;
  },
  async plans() {
    const res = await billingApi.getPlans();
    return res.data || [];
  },
  async changePlan(planKey) {
    const res = await billingApi.changePlan(planKey);
    return res.data;
  },
  async cancel() {
    const res = await billingApi.cancel();
    return res.data;
  },
  async invoices() {
    const res = await billingApi.getInvoices();
    return res.data || [];
  },
  async summary() {
    const res = await billingApi.getSummary();
    return res.data;
  },
};